src/main/rx/nrf24_syma.c

```diff
@@ -0,0 +1,293 @@
+            rcData[NRF24_AUX2] = PWM_RANGE_MAX;
+        }
+        if (packet[7] & 0x80) { // headless
+            rcData[NRF24_AUX1] = PWM_RANGE_MAX;
+        }
+    } else {
+        rcData[NRF24_PITCH] = convertToPwmSigned(packet[2]) ^ 0x80; // elevator
+        rcData[NRF24_YAW] = convertToPwmSigned(packet[1]); // rudder
+        if (packet[14] & 0x08) { // picture
+            rcData[NRF24_AUX4] = PWM_RANGE_MAX;
+        }
+        if (packet[14] & 0x10) { // video
+            rcData[NRF24_AUX5] = PWM_RANGE_MAX;
+        }
+        if (packet[14] & 0x01) { // flip
+            rcData[NRF24_AUX2] = PWM_RANGE_MAX;
+        }
+        if (packet[14] & 0x04) { // rate
+            rcData[NRF24_AUX3] = PWM_RANGE_MAX;
+        }
+    }
+}
+
+static void hopToNextChannel(void)
+{
+    // hop channel every second packet
+    ++packetCount;
+    if ((packetCount & 0x01) == 0) {
+        ++rfChannelIndex;
+        if (rfChannelIndex >= rfChannelCount) {
+            rfChannelIndex = 0;
+        }
+    }
+    NRF24L01_SetChannel(rfChannels[rfChannelIndex]);
+}
+
+// The SymaX hopping channels are determined by the low bits of rxTxAddress
+void setSymaXHoppingChannels(uint32_t addr)
+{
+    addr = addr & 0x1f;
+    if (addr == 0x06) {
+        addr = 0x07;
+    }
+    const uint32_t inc = (addr << 24) | (addr << 16) | (addr << 8) | addr;
+    uint32_t * const prfChannels = (uint32_t *)rfChannels;
+    if (addr == 0x16) {
+        *prfChannels = 0x28481131;
+    } else if (addr == 0x1e) {
+        *prfChannels = 0x38184121;
+    } else if (addr < 0x10) {
+        *prfChannels = 0x3A2A1A0A + inc;
+    } else if (addr < 0x18) {
+        *prfChannels = 0x1231FA1A + inc;
+    } else {
+        *prfChannels = 0x19FA2202 + inc;
+    }
+}
+
+/*
+ * This is called periodically by the scheduler.
+ * Returns NRF24_RECEIVED_DATA if a data packet was received.
+ */
+nrf24_received_t symaDataReceived(uint8_t *payload)
+{
+    nrf24_received_t ret = NRF24_RECEIVED_NONE;
+    switch (protocolState) {
+    case STATE_BIND:
+        if (NRF24L01_ReadPayloadIfAvailable(payload, payloadSize)) {
+            const bool bindPacket = checkBindPacket(payload);
+            if (bindPacket) {
+                ret = NRF24_RECEIVED_BIND;
+                protocolState = STATE_DATA;
+                // using protocol NRF24L01_SYMA_X, since NRF24L01_SYMA_X5C went straight into data mode
+                // set the hopping channels as determined by the rxTxAddr received in the bind packet
+                setSymaXHoppingChannels(rxTxAddr[0]);
+                // set the NRF24 to use the rxTxAddr received in the bind packet
+                NRF24L01_WriteRegisterMulti(NRF24L01_0A_RX_ADDR_P0, rxTxAddr, RX_TX_ADDR_LEN);
+                packetCount = 0;
+                rfChannelIndex = 0;
+                NRF24L01_SetChannel(rfChannels[0]);
+            }
+        }
+        break;
+    case STATE_DATA:
+        if (micros() > timeOfLastHop + hopTimeout) {
```

Comment:
Maybe the code should check for received packet first and then for hop timeout? But chance of receiving packet just before timeout is negligible ...